File: 0007-reverse-integer/0007-reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:
        sign = -1
        if x>=0 :
            sign = 1
        temp = x*sign
        res = 0
        while(temp>0):
            res = res*10+temp%10
            temp//=10
        if res < -2**31 or res > 2**31 - 1:
            return 0
        return res*sign

File: 0007-reverse-integer/0007-reverse-integer.java
class Solution {
    public int reverse(int x) {
        long res = 0;
        int sign = x>=0 ? 1 : -1;
        int temp = x*sign;
        
        while(temp>0){
            res = res*10+temp%10;
            temp/=10;
        }
        if(res<Integer.MIN_VALUE || res>Integer.MAX_VALUE) return 0;
        else return (int) sign*(int)res;
    }
}